File: 03-poo/06-herencia/java/soluciones/RetoSupervisor.java
public class RetoSupervisor {
    static class Empleado {
        protected final String nombre;
        Empleado(String nombre) { this.nombre = nombre; }
        String responsabilidad() { return "Ejecutar tareas"; }
    }
    static class Supervisor extends Empleado {
        Supervisor(String nombre) { super(nombre); }
        @Override String responsabilidad() { return "Coordinar al equipo"; }
    }
    public static void main(String[] args) { System.out.println(new Supervisor("Luis").responsabilidad()); }
}

File: 03-poo/06-herencia/scala/soluciones/RetoSupervisor.scala
class EmpleadoReto(val nombre: String):
  def responsabilidad: String = "Ejecutar tareas"
class SupervisorReto(nombre: String) extends EmpleadoReto(nombre):
  override def responsabilidad: String = "Coordinar al equipo"

@main def retoSupervisor(): Unit = println(new SupervisorReto("Luis").responsabilidad)
